src/button-group/button-group.js

```diff
@@ -0,0 +1,74 @@
+/*
+Copyright (c) 2018 Uber Technologies, Inc.
+
+This source code is licensed under the MIT license found in the
+LICENSE file in the root directory of this source tree.
+*/
+// @flow
+
+import React from 'react';
+
+import {KIND, SIZE, SHAPE} from '../button/index.js';
+import {getOverrides} from '../helpers/overrides.js';
+
+import {StyledRoot} from './styled-components.js';
+import type {PropsT} from './types.js';
+
+function isSelected(selected, index) {
+  if (!selected) {
+    return false;
+  }
+
+  if (Array.isArray(selected)) {
+    return selected.includes(index);
+  }
+
+  return selected === index;
+}
+
+export default function ButtonGroup(props: PropsT) {
+  const {overrides = {}} = props;
+  const [Root, rootProps] = getOverrides(overrides.Root, StyledRoot);
+
+  return (
+    <Root aria-label={props.ariaLabel} {...rootProps}>
+      {React.Children.map(props.children, (child, index) => {
+        if (!React.isValidElement(child)) {
+          return null;
+        }
+
+        return React.cloneElement(child, {
+          disabled: props.disabled,
```

Comment:
Wonder if we want to allow a child-specific `disabled` prop, and then use the value from the button group only if it's set to `true`. So a single button can be disabled.